pkg/volume/csi/csi_plugin.go

```diff
@@ -366,23 +406,61 @@ func (p *csiPlugin) ConstructVolumeSpec(volumeName, mountPath string) (*volume.S
 
 	klog.V(4).Info(log("plugin.ConstructVolumeSpec extracted [%#v]", volData))
 
-	fsMode := api.PersistentVolumeFilesystem
-	pv := &api.PersistentVolume{
-		ObjectMeta: meta.ObjectMeta{
-			Name: volData[volDataKey.specVolID],
-		},
-		Spec: api.PersistentVolumeSpec{
-			PersistentVolumeSource: api.PersistentVolumeSource{
-				CSI: &api.CSIPersistentVolumeSource{
-					Driver:       volData[volDataKey.driverName],
-					VolumeHandle: volData[volDataKey.volHandle],
+	var spec *volume.Spec
+	inlineEnabled := utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume)
+
+	if inlineEnabled {
+		mode := driverMode(volData[volDataKey.driverMode])
```

Comment:
If new kubelet is started with json files produced by old kubelet, then driverMode is going to be empty. It should fall into `persistentDriverMode` below.

Reply:
great catch. Thanks.

Reply:
fixed.